src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttributeList.cs

```diff
@@ -25,6 +25,20 @@ public TagHelperAttributeList()
         /// </summary>
         /// <param name="attributes">The collection to wrap.</param>
         public TagHelperAttributeList(IEnumerable<TagHelperAttribute> attributes)
+            : base (new List<TagHelperAttribute>(attributes))
```

Comment:
Testing is a big part of this. We use it extensively in our tests but also when trying to copy `TagHelperAttributeList`s among each other. For instance we do this in the `Script` and `Link` `TagHelper`s in MVC.

Ultimately, I'd like to leave this in for ease of use. Without it, it'd makes the class quite annoying to test (no more returning `new[] { ... }`), it'd be over 50 changes in our own tests even.